Android/src/org/droidplanner/android/fragments/widget/video/BaseUVCVideoWidget.kt

```diff
@@ -0,0 +1,297 @@
+import android.graphics.SurfaceTexture
+import android.hardware.usb.UsbDevice
+import android.os.Bundle
+import android.util.Log
+import android.view.*
+import android.widget.TextView
+import com.o3dr.services.android.lib.drone.attribute.AttributeEvent
+import com.o3dr.services.android.lib.drone.attribute.AttributeType
+import com.o3dr.services.android.lib.drone.property.State
+import com.serenegiant.usb.DeviceFilter
+import com.serenegiant.usb.USBMonitor
+import com.serenegiant.usb.UVCCamera
+import org.droidplanner.android.R
+import org.droidplanner.android.dialogs.UVCDialog
+import org.droidplanner.android.fragments.widget.TowerWidget
+import org.droidplanner.android.fragments.widget.TowerWidgets
+import java.util.concurrent.LinkedBlockingQueue
+import java.util.concurrent.ThreadPoolExecutor
+import java.util.concurrent.TimeUnit
+import android.graphics.Matrix
+import org.droidplanner.android.utils.prefs.DroidPlannerPrefs
+
+abstract class BaseUVCVideoWidget : TowerWidget(){
+
+    companion object {
+
+        private val filter = initFilter()
+
+        private fun initFilter(): IntentFilter {
+            val temp = IntentFilter()
+            temp.addAction(AttributeEvent.STATE_CONNECTED)
+            temp.addAction(AttributeEvent.STATE_ARMING)
+            return temp
+        }
+
+    }
+
+    protected val DEBUG = false
+
+    protected val TAG = "BaseUVCVideoWidget"
+
+    override fun getWidgetType() = TowerWidgets.UVC_VIDEO
+
+    // for thread pool
+    protected val CORE_POOL_SIZE = 1   // initial/minimum threads
+    protected val MAX_POOL_SIZE = 4    // maximum threads
+    protected val KEEP_ALIVE_TIME = 10 // time periods while keep the idle thread
+    protected val EXECUTER: ThreadPoolExecutor = ThreadPoolExecutor(CORE_POOL_SIZE, MAX_POOL_SIZE, KEEP_ALIVE_TIME.toLong(),
+            TimeUnit.SECONDS, LinkedBlockingQueue<Runnable>())
+
+    // for accessing USB and USB camera
+    protected var mUSBMonitor: USBMonitor? = null
+    protected var mUVCCamera: UVCCamera? = null
+    protected var isPreview:Boolean = false
+    protected var usbDevice: UsbDevice? = null
+    protected var mPreviewSurface: Surface? = null
+
+    //Aspect ratio
+    protected var aspectRatio_4_3: Float = 3f / 4f
+    protected var aspectRatio_16_9: Float = 9f / 16f
+    protected var aspectRatio_21_9: Float = 9f / 21f
+    protected var aspectRatio_1_1: Float = 1f / 1f
+    protected var aspectRatio: Float = aspectRatio_4_3
+
+
+
+    protected val receiver = object : BroadcastReceiver() {
+        override fun onReceive(context: Context, intent: Intent) {
+            when (intent.action) {
+                AttributeEvent.STATE_CONNECTED -> {
+                    startVideoStreaming()
+                }
+                AttributeEvent.STATE_ARMING -> {
+                    val droneState = drone.getAttribute<State>(AttributeType.STATE)
+                    if (droneState.isArmed){
+                        startVideoStreaming()
```

Comment:
@guiseco since you're streaming on `STATE_CONNECTED` above, I'd assume you'd also want to stream here regardless of the arming status?